pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+
+### Custom Scripts Wrappers
+
+Almost every interpreter has its own way of doing the same thing needed for most common use cases, and with this in mind we abstracted most operations to allow a terser *core* for anyone to consume, granting that its functionalities are the same, no matter which interpreter one prefers.
+
+There are also cases that are not tackled directly in *core*, but necessary to anyone trying to extend *core* as it is, so that some helper felt necessary to enable users and contributors as much as they want.
+
+In few words, while every *interpreter* is literally passed along to unlock its potentials 100%, the most common details or operations we need in core are:
+
+| name                      | example                                       | behavior |
+| :------------------------ | :-------------------------------------------- | :--------|
+| type                      | `wrap.type`                                   | Return the current `type` (interpreter or custom type) used in the current code execution. |
+| interpreter               | `wrap.interpreter`                            | Return the *interpreter* _AS-IS_ after being bootstrapped by the desired `config`. |
+| XWorker                   | `wrap.XWorker`                                | Refer to the global `XWorker` class available to the main thread code while executing. |
+| io                        | `wrap.io`                                     | Allow to lazily define different `stdout` or `stderr` via the running *interpreter*. This `io` field can be lazily defined and restored back for any element currently running the code. |
+| config                    | `wrap.config`                                 | It is the resolved *JSON* config and it is an own clone per each element running the code, usable also as "_state_" reference for the specific element, as changing it at run time will never affect any other element. |
+| run                       | `wrap.run(code)`                              | It abstracts away the need to know the exact method name used to run code synchronously, whenever the *interpreter* allows such operation, facilitating future migrations from an interpreter to another. |
+| runAsync                  | `wrap.run(code)`                              | It abstracts away the need to know the exact method name used to run code asynchronously, whenever the *interpreter* allows such operation, facilitating future migrations from an interpreter to another. |
+
+This is the `wrap` mentioned with most hooks and initializers previously described, and we're more than happy to learn if we are not passing along some extra helper.
```

Comment:
All of the above is absolutely amazing! I think that adding an example of these would go a looooooong way to help users understand how they can be used (Yes, these are advanced feature but, still, would be great)